### PR TITLE
script to test end of year closure in date picker

### PR DESCRIPTION
**Pull request by agnesgaroux**
## What does this change?

The logic that generates dates for the date picker dropdown (items request) now live across applications.
This script can be run to easily check what dates will be listed on any given day

## How to test
Set `dateNow` as the date you want to test
Run `yarn check_holiday_closures`

## How can we measure success?

We're confident that the date picker will display correct dates over the holiday closure

## Have we considered potential risks?

If the logic changes in the catalogue-api, this script will need updating too

### CODE REVIEW

File: api/scripts/holiday_closure_test.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    : '';
+
+const compareDates = (library: NextOpeningDate, deepstore: NextOpeningDate) => {
+  const libraryDate =
+    library.open && DateTime.fromJSDate(new Date(library.open)).startOf('day');
+  const deepstoreDate =
+    deepstore.open &&
+    DateTime.fromJSDate(new Date(deepstore.open)).startOf('day');
+  if (libraryDate === undefined || deepstoreDate === undefined) {
+    throw 'One of these dates is undefined!';
```
Comment (rcantin-w) [inline api/scripts/holiday_closure_test.ts:28]: I'm getting an eslint error here (it should've been caught automatically but I guess there's still improvements to do on this repo!)

> Expected an error object to be thrown.
> https://eslint.org/docs/latest/rules/no-throw-literal

File: api/scripts/holiday_closure_test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+// set dateNow in utils.ts as the date to be tested and run this script with
+// yarn check_holiday_closures
+
+import { DateTime } from 'luxon';
+import { getElasticClient } from '@weco/content-common/services/elasticsearch';
+import { ElasticsearchVenue, Venue } from '@weco/content-common/types/venue';
+import { getNextOpeningDates } from '@weco/content-api/src/controllers/utils';
+import { NextOpeningDate } from '@weco/content-common/types/venue';
+
+// set this to the date to be tested
```
Comment (rcantin-w) [inline api/scripts/holiday_closure_test.ts:13]: Could you add an example here/a commented out line, e.g.
```
// const dateNow = new Date('2023-02-39........');
```

Comment (rcantin-w) [reply api/scripts/holiday_closure_test.ts:13]: (to ensure it's the correct format)

**Review by rcantin-w — COMMENTED** [review]
I _might_ be doing it wrong, but whatever date I put in I get dates in December and January?
Screenshot 2024-12-10 at 10 58 52
Screenshot 2024-12-10 at 11 03 04

**Comment by rcantin-w** [comment]
This will be super useful to have, thank you again for taking this on - note that all changes I ask about above can fully wait since you've already confirmed the closures will be OK this year!

**Comment by gestchild** [comment]
Really useful, thank you!

Ran into the same thing as @rcantin-w where I needed to set dateNow in util.ts and holiday_closure_test.ts, but then it works fine.

**Comment by rcantin-w** [comment]
Ahhh yes thank you @gestchild !
Screenshot 2024-12-10 at 13 37 42
Screenshot 2024-12-10 at 13 40 27

So probably worth pointing out in the test instructions.

**Review by rcantin-w — APPROVED** [review]

**Pull request merged**